src/components/ArticleList.jsx

```diff
@@ -2,32 +2,42 @@ import { useEffect, useState } from "react";
 import * as Api from "../utils/newsApi";
 import { ArticleCard } from "./ArticleCard";
 
-export const ArticleList = () => {
+export const ArticleList = ({topics}) => {
   const [articleList, setArticleList] = useState([]);
   const [isLoading, setIsLoading] = useState(true)
+console.log(topics)
 
   useEffect(() => {
     setIsLoading(true)
     Api.getArticles().then(({ articles }) => {
-      setArticleList(articles);
+      if(topics !== ""){
+        setArticleList(articles.filter((article)=> article.topic === topics))
+         }else{
```

Comment:
This is filtering the list "Client Side" - the backend has an endpoint query parameter to fetch a filtered list which is intended to be utilised on this ticket